great_expectations/execution_engine/pandas_execution_engine.py

```diff
@@ -148,12 +136,9 @@ def _instantiate_azure_client(self) -> None:
                 pass
 
     def _instantiate_s3_client(self) -> None:
-        # Try initializing cloud provider client. If unsuccessful, we'll catch it when/if a BatchSpec is passed in.
+        # Try initializing cloud provider client. If unsuccessful, we'll die here.
         boto3_options = self.config.get("boto3_options", {})
-        try:
-            self._s3 = boto3.client("s3", **boto3_options)
-        except (TypeError, AttributeError):
-            self._s3 = None
+        self._s3 = aws.boto3.client("s3", **boto3_options)
```

Comment:
@Shinnnyshinshin Has this been tested locally with all aws/s3/boto/botocore packages uninstalled?  If not, then please consider running the tests of the functionality that exercises `_instantiate_s3_client()` (while these components are uninstalled) to be certain that they fail gracefully.  Thanks.

Reply:
Hi @alexsherstinsky  So in our tests we actually don’t hit the line in `_instantiate_s3_client()`
they are all mocked, so we error-out earlier at the Mock step. 

```python 
>   import botocore
E   ModuleNotFoundError: No module named 'botocore'

../ENVs/supercon_ge/lib/python3.9/site-packages/moto/core/models.py:1: ModuleNotFoundError

```

Reply:
@Shinnnyshinshin But in real life, there will be no mock -- do we need an integration test?

Reply:
@Shinnnyshinshin I see a potential issue here -- if we do not check for `boto3` to be "Truthy".  Thanks.

Reply:
In a real world setting, we would hit the error much earlier. 

1. In the `fluent` world, we would be prevented from adding the Datasiource to context with the following error

```
datasource = context.sources.add_or_update_pandas_s3(
    name="s3_datasource", bucket="taxi-data-sample-test"
)

# would give 

TestConnectionError: Attempt to connect to datasource failed with the following error message: Unable to create "PandasS3Datasource" due to missing boto3 dependency.

```

2. In the non-fluent world, we would hit it at `gx.get_context()` with the following message: 

```
 No module named 'boto3'
Cannot initialize datasource my_datasource: Cannot initialize datasource my_datasource, error: Unable to load boto3 (it is required for InferredAssetS3DataConnector).
```

Reply:
In either case, we wouldn't get to the ExecutionEngine. If we were there, we would want it to die anyway